.travis.yml

```diff
@@ -1,33 +1,72 @@
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=focal
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=6
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=7
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=8
+  # Testing Xilinx 7 Series toolchain on all OSes
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=trusty
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=xenial
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=bionic
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=eoan
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=focal
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=6
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=7
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=8
 
-      script:
-            - tuttest README.md unnamed3 | bash -
-    - #
-      install:
-            - tuttest README.md unnamed2 | bash -
+install:
+ - if [ "$OS" = "ubuntu" ]; then export INSTALL_COMMAND="apt update && apt install -y"; fi
+ - if [ "$OS" = "centos" ]; then export INSTALL_COMMAND="yum -y install"; fi
+ - if [ "$OS" = "ubuntu" ]; then export INSTALL_PACKAGES="locales software-properties-common git wget xz-utils make build-essential zlib1g libz-dev libffi-dev ca-certificates"; fi
+ - if [ "$OS" = "centos" ]; then export INSTALL_PACKAGES="git wget make gcc-c++ zlib-devel xz libffi-devel perl-Pod-Html diffutils byacc"; fi
+ - export DOCKER_NAME=test
+ - export IN_DOCKER_EXEC="docker exec -it $DOCKER_NAME"
+ - docker create
+        --name $DOCKER_NAME
+        --env LANG="en_US.UTF-8"
+        --env LANGUAGE="en_US"
+        --env DEBIAN_FRONTEND="noninteractive"
+        -v /home/travis:/home/travis
+        --workdir $PWD
+        --interactive
+        --tty
+        $OS:$OS_VERSION
+ - docker start $DOCKER_NAME
+ - $IN_DOCKER_EXEC bash -c "$INSTALL_COMMAND $INSTALL_PACKAGES"
+ - if [ "$OS" = "ubuntu" ]; then $IN_DOCKER_EXEC locale-gen en_US.UTF-8; fi
+ - if [ "$OS" = "ubuntu" ]; then $IN_DOCKER_EXEC update-locale; fi
+ - $IN_DOCKER_EXEC wget https://www.openssl.org/source/openssl-1.1.1c.tar.gz
+ - $IN_DOCKER_EXEC tar xf openssl-1.1.1c.tar.gz
+ - $IN_DOCKER_EXEC bash -c 'cd openssl-1.1.1c && ./config --prefix=/usr/local --openssldir=/usr/local/ssl shared zlib && make install > /dev/null'
+ - if [ "$OS" = "centos" ]; then $IN_DOCKER_EXEC bash -c 'echo "/usr/local/lib64" | tee -a /etc/ld.so.conf'; fi
+ - $IN_DOCKER_EXEC ldconfig
+ - if [ "$OS" = "centos" ] && [ "$OS_VERSION" == "8" ]; then $IN_DOCKER_EXEC bash -c "wget https://kerberos.org/dist/krb5/1.17/krb5-1.17.1.tar.gz && tar xf krb5-1.17.1.tar.gz && cd krb5-1.17.1/src && ./configure && make && cp lib/libk5crypto.so.3.1 /lib64/"; fi
+ - if [ "$OS" = "centos" ] && [ "$OS_VERSION" == "8" ]; then $IN_DOCKER_EXEC bash -c "rm /lib64/libk5crypto.so.3 && ln -s /lib64/libk5crypto.so.3.1 /lib64/libk5crypto.so.3"; fi
+ - $IN_DOCKER_EXEC ldconfig
+ - $IN_DOCKER_EXEC wget https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tar.xz --ca-directory=/etc/ssl/certs/
```

Comment:
Why are you installing Python like this?

Reply:
See above. Tuttest requires python 3.7, but it is not available on official ubuntu repositories and one from unofficial ones is missing required features.

Reply:
@kamilrakoczy We shouldn't be installing `tuttest` inside the docker container, instead we should extend tuttest as described at https://github.com/antmicro/tuttest/issues/10

Reply:
@mithro I added this option to ``tuttest`` and updated PR.